File: src/Python/Preprocessing/gaussnodes.py
import os
import numpy

def gaussnodes(pgauss,dim,elemtype):

    d0 = os.getcwd();
    ii = d0.find("Exasim");
    fn = d0[0:ii] + "Exasim/src/Python/Preprocessing/gaussnodes.bin";
    tmp = numpy.fromfile(open(fn, "r"), dtype=numpy.float64);

    ndims = numpy.int_(tmp[0]);
    k1 = 1;
    k2 = k1+(ndims);
    narrays = numpy.int_(tmp[k1:k2]);
    k1 = k2; k2 = k1 + numpy.prod(narrays);
    sz1 = numpy.reshape(numpy.int_(tmp[k1:k2]), narrays, order='F');
    k1 = k2; k2 = k1 + numpy.prod(narrays);
    sz2 = numpy.reshape(numpy.int_(tmp[k1:k2]), narrays, order='F');
    sz = numpy.reshape(sz1*sz2, [numpy.prod(narrays), 1], order='F');
    lz = numpy.cumsum(numpy.insert(sz, 0, 0, axis=0));
    n = len(lz);
    lz1 = numpy.reshape(lz[0:(n-1)], narrays, order='F');
    lz2 = numpy.reshape(lz[1:n], narrays, order='F');

    e = elemtype;
    pm1 = pgauss-1;
    dm1 = dim-1;

    i = 0;
    m1 = k2 + lz1[i,e,pm1,dm1];
    m2 = m1 + lz2[i,e,pm1,dm1]-lz1[i,e,pm1,dm1];
    xgauss = numpy.reshape(tmp[m1:m2],[sz1[i,e,pm1,dm1],sz2[i,e,pm1,dm1]],order='F');

    i = 1;
    m1 = k2 + lz1[i,e,pm1,dm1];
    m2 = m1 + lz2[i,e,pm1,dm1]-lz1[i,e,pm1,dm1];
    wgauss = numpy.reshape(tmp[m1:m2],[sz1[i,e,pm1,dm1],sz2[i,e,pm1,dm1]],order='F');    

    return xgauss, wgauss;

File: Version0.2/Matlab/Preprocessing/gaussnodes.m
function [xgauss, wgauss] = gaussnodes(pgauss,dim,elemtype)

fileID = fopen("gaussnodes.bin",'r');
tmp = fread(fileID,'double');
fclose(fileID);

ndims = tmp(1);
k1 = 2; k2 = (k1-1)+(ndims);
narrays = tmp(k1:k2);
k1 = k2+1; k2 = (k1-1) + prod(narrays);
sz1 = reshape(tmp(k1:k2), narrays(:)');
k1 = k2+1; k2 = (k1-1) + prod(narrays);
sz2 = reshape(tmp(k1:k2), narrays(:)');
sz = sz1.*sz2;
lz = cumsum([0; sz(:)]);
lz1 = reshape(lz(1:end-1),narrays(:)');
lz2 = reshape(lz(2:end),narrays(:)');
      
e = elemtype+1;

i = 1;
m1 = k2+1 + lz1(i,e,pgauss,dim);
m2 = m1-1 + lz2(i,e,pgauss,dim)-lz1(i,e,pgauss,dim);
xgauss = reshape(tmp(m1:m2),sz1(i,e,pgauss,dim),sz2(i,e,pgauss,dim));

i = 2;
m1 = k2+1 + lz1(i,e,pgauss,dim);
m2 = m1-1 + lz2(i,e,pgauss,dim)-lz1(i,e,pgauss,dim);
wgauss = reshape(tmp(m1:m2),sz1(i,e,pgauss,dim),sz2(i,e,pgauss,dim));


